pegasus/helpers/form_helpers.rb

```diff
@@ -56,25 +56,25 @@ def insert_or_upsert_form(kind, data, options={})
 
   # For HOC signups, a duplicate (matching email and name) entry is assumed to be an update rather
   # than an insert.
-  num_rows_updated = 0
-  if (kind =~ /HocSignup/) == 0
+  if kind.start_with? 'HocSignup'
+    row_for_update = row.dup
+    [:created_at, :created_ip, :secret].each {|key| row_for_update.delete key}
     num_rows_updated = DB[:forms].
-      where(email: row[:email], kind: kind, name: row[:name]).
-      update(row)
+      where(kind: kind, email: row_for_update[:email], name: row_for_update[:name]).
+      update(row_for_update)
+    return row_for_update if num_rows_updated > 0
```

Comment:
I know this is out of scope for this PR, but I think this would be a lot cleaner with a refactor to generalize. 
The form helpers shouldn't know or care about any specific form types, let alone make decisions based on that. 

What if instead we (optionally) implement a set of methods, say `should_update?(data)` and `update(data)` which we will call if they exist for the form class? Note we already [have](https://github.com/code-dot-org/code-dot-org/blob/1bbf2fe5fd3e807206bf52c8afceb709d5a5c7df/pegasus/helpers/form_helpers.rb#L55) the form class and check if methods exist. Or even just the `should_update?`, with only HOC forms implementing it and always returning true.